app/controllers/vitals_controller.rb

```diff
@@ -59,6 +59,7 @@ def destroy
   # For chart actions
   def chart
     @selected_vital = current_user.vitals.find_by(id: params[:vital_id])
+    authorize @selected_vital
```

Comment:
If there's no distinction between @selected_vital and @vital aside from what you decided to name the variable, you could just adopt the standard naming convention and use @vital in line 61 and change your html.erb to reference @vital instead of @selected_vital. Using @vital in the chart method would allow you to continue to use an all-inclusive before_action without needing to create an exception and calling a separate authorize here since all of them use the same policy.

Reply:
Great feedback thank you. Changed routes to accommodate this refactor. Code is looking better I think 😃